woocommerce/woocommerce
# [test reports] Optimize the reporting job - don't duplicate reports and remove Allure report

### Submission Review Guidelines:

### Changes proposed in this Pull Request:

This fixes duplicated reports when a job is re-run. What happens is the test-reports job is re-run whenever any test job is re-run and even if the tests that have a report are not actually running the second time. Example: if only one blocks e2e job is re-run the test reports for the core e2e tests are recreated, using the previous run artifacts.

I also removed the steps that generate the Allure report. It doesn't bring too much value having it archived, because one needs to have allure cli installed and need to run an allure command (`allure open`) anyway to open the report locally, so might as well run the `generate` or simply `serve` command.

### How to test the changes in this Pull Request:

In CI:
See [this run](https://github.com/woocommerce/woocommerce/actions/runs/9581127342/attempts/1)
- Have a test run with multiple reports (e.g. core-api-tests, core-e2e-tests)
- After all tests finished, artifacts are uploaded under the `artifact-attempt-N` name pattern (e.g. core-api-tests-attempt-1, core-e2e-tests-attempt-2).
- The 'Publish report to dashboard' step runs and sends the correct artifact name in the workflow call

See [this run](https://github.com/woocommerce/woocommerce/actions/runs/9581127342).
- Re-run one job (e.g. api tests)
- New artifacts are created only for the re-run report name (e.g. core-api-tests-attempt-2)
- The 'Publish report to dashboard' step runs only for the new artifacts and is skipped for the rest
